cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -114,7 +114,13 @@ struct row_group_info {
  */
 struct metadata : public FileMetaData {
   metadata() = default;
-  explicit metadata(datasource* source);
+  explicit metadata(datasource* source, bool read_page_indexes = true);
+  metadata(metadata const& other)            = delete;
```

Comment:
This performance issue is probably a good reminder to be explicit about which classes we want to be copyable or move-only, etc.